#Relación de Dispersión

La relación de dispersión obtenida en la teoría lineal de olas está dada por:

$\omega^2 = g k \tanh(kh)$

donde $\omega$ es la frecuencia angular, $g$ la aceleración de la gravedad, $k$ el número de onda y $h$ la profundidad.



###Resolviendo la relación de dispersión
Definiendo $x = kh$ y $\beta = \frac{\omega^2 h}{g}$ puedo escribir la relación de dispersión como:

$\frac{\beta}{x} = \tanh(x)$

Con esta expresión, puedo aplicar un método iterativo para encontrar $x$.

Para el método de Newton-Raphson, defino: $f(x) = \tanh(x) - \frac{\beta}{x}$ y $f'(x) =  1-\tanh^2 (x) + \frac{\beta}{x^2}$

Luego, puedo usar la expresión iterativa:

$x_{n+1} = x_{n}-\frac{f(x_{n})}{f'(x_{n})}$

y resolver la ecuación.

Es necesario un valor inicial $x_{0}$ y determinar la precisión que queremos para nuestra solución.

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
#@title Defino "longitud" :Función que obtiene L resolviendo la relación de dispersión con el método de N-R
beta=lambda h,T:h*(2*np.pi/T)**2/9.81
f=lambda x,b: np.tanh(x)-b/x
df=lambda x,b: (1-np.tanh(x)**2)+b/(x**2)
def longitud(h,T,x0,ex):
  be=beta(h,T)
  x1=x0-f(x0,be)/df(x0,be)
  paso=[]
  paso.append(x0)
  while abs(x1-x0)>ex:
    x0=x1
    x1=x1-f(x1,be)/df(x1,be)
    paso.append(x1)
  k=x1/h
  L=2*np.pi/k
  return L, paso

La función **longitud(h,T,x0,ex)** está definida de manera tal que devuelve *L* y también un array con todos los valores de x=k*h que fue encontrando hasta llegar a la solución.

In [ ]:
L,iteraciones=longitud(5,15,0.1,0.0001)
L,len(iteraciones)

(103.4855879990278, 5)

Si quiero calcular L para todos los valores de h y T que me piden en el TP:

In [ ]:
h=[5,20,100]
T=np.arange(1,16,1)
ele=np.empty((len(T),len(h)))
pasos=np.empty((len(T),len(h)))
for i in range(len(h)):
  for j in range(len(T)):
    L,iter=longitud(h[i],T[j],0.1,0.0001)
    ele[j,i]=L
    pasos[j,i]=len(iter)
    del (L,iter)

In [ ]:
print(ele)
print(pasos)

[[  1.56130999   1.56130999   1.56130999]
 [  6.24470681   6.24523997   6.24523997]
 [ 13.76238574  14.05178945  14.05178993]
 [ 22.19816867  24.97882724  24.98095987]
 [ 30.30922666  38.91067283  39.03274979]
 [ 38.08973813  55.04954297  56.20715968]
 [ 45.65524086  71.98212929  76.50417835]
 [ 53.08152309  88.79267466  99.92314924]
 [ 60.41394923 105.211631   126.45388637]
 [ 67.68045433 121.23690666 156.03175792]
 [ 74.8989873  136.93096951 188.43922692]
 [ 82.08155639 152.35895252 223.22011783]
 [ 89.23648621 167.57488453 259.71529824]
 [ 96.36970891 182.62096346 297.22033128]
 [103.485588   197.52967805 335.14390663]]
[[13. 15. 17.]
 [11. 13. 15.]
 [ 9. 11. 14.]
 [ 8. 11. 13.]
 [ 8. 10. 12.]
 [ 7.  9. 12.]
 [ 7.  9. 11.]
 [ 6.  8. 11.]
 [ 6.  8. 11.]
 [ 6.  8. 10.]
 [ 6.  7. 10.]
 [ 6.  7. 10.]
 [ 5.  7.  9.]
 [ 5.  7.  9.]
 [ 5.  7.  9.]]


###Algunas aproximaciones

*Eckart (1951)*

$\omega^{2} = gk\sqrt{\tanh (\frac{\omega^2}{g} h)}$

*Hunt (1979)*

$(kh)^2 = y^2 + \frac{y}{1+\sum_{n=1}^6 d_{n}y^n}$

con $y=\omega^2h/g = k_{0}h$ y $d_{1}=0.6666666666$, $d_{2}=0.3553553553$, $d_{3}=0.1608465608$, $d_{4}=0.0632098765$, $d_{5}=0.0217540484$ y $d_{6}=0.0065407983$

*Guo (2002)* propuso la formulación:

$L = \frac {2\pi h}{x^2(1-e^{-x^\beta})^{-\frac{1}{\beta}}} $ donde $x = \frac{ h\omega}{\sqrt{gh}}$ y $\beta = 2.4908$

## Otras formas de escribir la relación de dispersión

$\omega^2 = gk \tanh (kh)$

$C^2 = (\frac{L}{T})^2 = \frac{g}{k} \tanh (kh)$

$L= \frac{g}{2\pi} T^2 \tanh(kh)$

Y la útlima podría reescribise, tomando $L_{o} = \frac{g}{2\pi} T^2 = 1.56 T^2$ [**Longitud de onda en AP**]

$L = L_{o} \tanh (kh)$

$C = C_{o} \tanh (kh)$

